### PR TITLE
[MRG] release pr

### PR DESCRIPTION
**Pull request by jasmainak**
closes https://github.com/autoreject/autoreject/issues/100

### CODE REVIEW

File: doc/api.rst [hunk 1]
```diff
@@ -14,8 +14,6 @@ Classes
 .. autosummary::
    :toctree: generated/
 
-   GlobalAutoReject
-   LocalAutoReject
```
Comment (agramfort) [inline doc/api.rst]: make LocalAutoReject private?

Comment (jasmainak) [reply doc/api.rst]: yes! If people want they can still pass a list of single element to `LocalAutoRejectCV` ...

File: examples/plot_global_reject.py [hunk 2]
```diff
@@ -50,10 +50,10 @@
 # (RMSE) values at the candidate thresholds, we will use the function
 # :func:`autoreject.validation_curve`.
 
-from autoreject import GlobalAutoReject, validation_curve  # noqa
+from autoreject import _GlobalAutoReject, validation_curve  # noqa
```
Comment (jasmainak) [inline examples/plot_global_reject.py]: @agramfort @dengemann we cannot expose private classes / methods in examples. What do you suggest for this?

Comment (jasmainak) [reply examples/plot_global_reject.py]: I suppose we can remove this example. It is reproducing Figure 1 in our paper ...

**Comment by agramfort** [comment]
well replicating figure from paper is interesting no? any middle ground?

**Comment by jasmainak** [comment]
> well replicating figure from paper is interesting no? any middle ground?

we could init the private class inside the `validation_curve` function and not pass any estimator to it. It's a bit different from sklearn API but I could live with it ...

**Comment by jasmainak** [comment]
okay, this is good to go on my end @agramfort @dengemann 

let's merge and release !

File: autoreject/utils.py [hunk 3]
```diff
@@ -10,6 +10,9 @@
 from mne.utils import check_version as version_is_greater_equal
 from mne import pick_types, pick_channels, pick_info
 from mne.channels.interpolation import _do_interp_dots
+from mne.externals import six
+
+import numpy as np
```
Comment (agramfort) [inline autoreject/utils.py]: don't import np after mne

**Comment by jasmainak** [comment]
> Why is _AutoReject private now? 

because 1) It does not support multi-channel fit 2) You can do what it does by passing a list of single element.

> And: if we have only one public class now why call it AutoRejectCV?

I would keep it like that for programming sanity so we have the distinction between the two types of classes in our codebase

**Comment by dengemann** [comment]
Can we use AutoRejectCV to reject data given we set the reject log, kappa
and rho? E.g. after round trip to disk?

From a user stand point AutoReject is easier to memorize and does not pose
the question of what the CV is good for. I’d prefer that one. Bit if it’s
only me who cares I won’t make trouble.

**Comment by jasmainak** [comment]
> Can we use AutoRejectCV to reject data given we set the reject log, kappa and rho? E.g. after round trip to disk?

yes you can set `rho` and `kappa` but `reject log` would need to be recomputed. I think this is the same with `AutoReject`, no?

> From a user stand point AutoReject is easier to memorize and does not pose the question of what the CV is good for. I’d prefer that one. Bit if it’s only me who cares I won’t make trouble.

So, do you suggest two classes `AutoReject` and `_AutoReject` to replace the older classes `LocalAutoRejectCV` and `LocalAutoReject` ? I don't know what @agramfort thinks. I can live with both options :)

**Comment by agramfort** [comment]
@dengemann what is your suggestion?

**Comment by dengemann** [comment]
Think of the busy (not necessarily sklearn) user who wants to be as little
confused as possible. He does not care abou CV and ways of doing things. He
wants what we describe in the paper. So my suggestion would be to call the
public class and call it AutoReject. The private class would be _AutoReject
or _AutorejectLocal or whatever we need to call it to make coding life easy.

My 2 cents.

**Comment by agramfort** [comment]
that's ok with me. +1

**Comment by jasmainak** [comment]
okay comments addressed

File: autoreject/utils.py [hunk 4]
```diff
@@ -105,6 +108,53 @@ def set_matplotlib_defaults(plt, style='ggplot'):
...
+
+    Parameters
+    ----------
+    params : dict
+        The dictionary to pretty print
+    offset : int
+        The offset in characters to add at the begin of each line.
+    printer : callable
+        The function to convert entries to strings, typically
+        the builtin str or repr
```
Comment (agramfort) [inline autoreject/utils.py]: returns?

**Comment by dengemann** [comment]
I love it! It looks beautiful.

**Comment by agramfort** [comment]
great @jasmainak 

can you update the online doc so we can see how it looks? thanks

**Comment by jasmainak** [comment]
website update !

http://autoreject.github.io/

**Comment by dengemann** [comment]
Readme shows old code ...

**Comment by jasmainak** [comment]
updated in this PR: https://github.com/autoreject/autoreject/pull/98

**Pull request merged**